src/lib/operators/abstract_aggregate_operator.hpp

```diff
@@ -90,19 +93,47 @@ class AggregateFunctionBuilder<ColumnDataType, AggregateType, AggregateFunction:
   }
 };
 
+template <typename ColumnDataType, typename AggregateType>
+class AggregateFunctionBuilder<ColumnDataType, AggregateType, AggregateFunction::StdDevSamp> {
+ public:
+  auto get_aggregate_function() {
+    return [](const ColumnDataType& new_value, std::optional<AggregateType>& current_primary_aggregate,
+              std::optional<AggregateType>& current_secondary_aggregate) {
+      if constexpr (std::is_arithmetic_v<ColumnDataType>) {
+        // add new value to sum of x
+        if (current_primary_aggregate) {
+          *current_primary_aggregate += new_value;
+        } else {
+          current_primary_aggregate = new_value;
+        }
+        // add new value to sum of x²
+        if (current_secondary_aggregate) {
+          *current_secondary_aggregate += new_value * new_value;
+        } else {
+          current_secondary_aggregate = new_value * new_value;
+        }
+      } else {
```

Comment:
Can we ever reach this branch? This [check](https://github.com/hyrise/hyrise/pull/1684/files#diff-d12958a0e19777d00bbb31d51731b7a4R68) should make it impossible, right?

Reply:
That's right. But we still have to do the `is_arithmetic` check here, since `ColumnDataType` could be string. So if the type is string, `new_value * new_value` would be invalid.

Is there a way to restrict the type on the template-type level?

Reply:
`if constexpr(...) {....} else { Fail("this shouldn't be reached"); }` is how I'd do it as well. Done like this in a couple of places.